### PR TITLE
Exclude IAM optional groups from VOMS AC

### PR DESCRIPTION
**Pull request by rmiccoli**
In the context of WLCG profile, this PR removes from VOMS attribute certificate the groups those have "wlcg.optional-group" label set.

### CODE REVIEW

**Review by enricovianello — APPROVED** [review]

File: iam-voms-aa/src/test/java/it/infn/mw/voms/VomsAcTests.java [hunk 1]
```diff
@@ -243,6 +243,32 @@ public void allGroupsAreReturnedForUser() throws Exception {
...
+
+    VOMSResponse response = parser.parse(new ByteArrayInputStream(xmlResponse));
+    assertThat(response.hasErrors(), is(false));
+    VOMSAttribute attrs = getAttributeCertificate(response);
+    assertThat(attrs.getFQANs(), hasSize(2));
+    assertThat(attrs.getFQANs(), hasItem("/test"));
+    assertThat(attrs.getFQANs(), hasItem("/test/sub"));
+    assertThat(attrs.getNotAfter(), lessThanOrEqualTo(Date.from(NOW_PLUS_12_HOURS)));
+  }
+
```
Comment (giacomini) [inline iam-voms-aa/src/test/java/it/infn/mw/voms/VomsAcTests.java:272]: And what about a situation of a role in an optional group? e.g. what happens (or should happen) if you ask the role but don't ask for the optional group?

Comment (rmiccoli) [reply iam-voms-aa/src/test/java/it/infn/mw/voms/VomsAcTests.java:272]: For now, if you have, for instance
* root group `/test`
* optional group `/optional` (subgroup of the root group)
* role `VO-Admin` in the optional group
asking for `/test/optional/Role=VO-Admin` returns two fqans, `/test` and `/test/optional/Role=VO-Admin`.

Comment (giacomini) [reply iam-voms-aa/src/test/java/it/infn/mw/voms/VomsAcTests.java:272]: I wonder if it should return also the fqan `/test/optional`. It's strange to have a role in a group but not that group.

Comment (rmiccoli) [reply iam-voms-aa/src/test/java/it/infn/mw/voms/VomsAcTests.java:272]: Makes sense, but the same applies to tokens. If we have a subgroup called `sub` within an optional group `root`, when we request only `/root/sub`, we get `/root/sub` but not `/root`.

Comment (giacomini) [reply iam-voms-aa/src/test/java/it/infn/mw/voms/VomsAcTests.java:272]: Optional groups confirm to be a source of confusion and ambiguity. Leave it as is for the moment.

**Review by enricovianello — APPROVED** [review]

**Pull request merged**